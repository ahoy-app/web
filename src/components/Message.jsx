import React from 'react'
import PropTypes from 'prop-types'

import { color, shadow } from '../style'
import T from './Text'
import Spacer from './Spacer'

const styles = {
  background: color.background.dark,

  minHeight: '20px',
  width: '500px',

  padding: '5px 10px 5px 10px',
  boxSizing: 'border-box',

  display: 'flex',
  flexDirection: 'column',

  borderRadius: '10px',
  filter: shadow.light,
}

const Message = ({ message, me }) => (
  <div
    style={{
      ...styles,
      background: me ? color.background.color : color.background.light,
    }}
  >
    <T.body>{message.from}</T.body>
    <Spacer bottom={5} />
    <T.body>{message.content}</T.body>
  </div>
)

Message.propTypes = {
  message: PropTypes.object.isRequired,
  me: PropTypes.bool.isRequired,
}

export default Message
