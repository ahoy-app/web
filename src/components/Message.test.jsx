import React from 'react'
import { shallow } from 'enzyme'

import Message from './Message'

const message = {
  id: 'q384tp1349fgh139p48g',
  from: 'Ted',
  content: 'See you later',
}

describe('Message', () => {
  it('should render correctly', () => {
    const component = shallow(<Message message={message} me={false} />)
    expect(component).toMatchSnapshot()
  })
})
