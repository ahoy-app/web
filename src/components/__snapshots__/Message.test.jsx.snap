// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "borderRadius": "10px",
      "boxSizing": "border-box",
      "display": "flex",
      "filter": "drop-shadow(2px 5px 2px #0003)",
      "flexDirection": "column",
      "minHeight": "20px",
      "padding": "5px 10px 5px 10px",
      "width": "500px",
    }
  }
>
  <Text-body>
    Ted
  </Text-body>
  <Spacer
    bottom={5}
  />
  <Text-body>
    See you later
  </Text-body>
</div>
`;
